Move events stubs to events dir

rajkan01 opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->
fixes #14791 

- This PR makes the mbed-stubs-events library more self-contained by removing the dependencies on the monolithic header libraries we created to make implementing the stubs easier. We also move the files to the events directory so the stubs live with the associated component.
- Add mbed-headers-events library inside CMake of doubles
- Move events stubs headers from mbed-stubs-headers library to mbed-stubs-events library.
- Updated connectivity unit test CMake to use mbed-stubs-events-headers as it relies on events stub headers


<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

#### Impact of changes <!-- Optional -->
None.
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
None
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->
To be updated
<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

rajkan01 committed:
CMake: Move events stubs sources to the mbed-os/events directory

Move the events stub into the events component directory.
so we can avoid duplicating the mbed-os source tree in a central
UNITTESTS folder.

rajkan01 committed:
CMake: unit-tests: Move mbed-headers-events to the events directory

Move the header-only mbed-headers-events library the unit test stubs
depend on into the events component directory. This makes the events
stubs more self-contained and improves the composition of the library.

rajkan01 committed:
CMake: Move events stub headers to the events/tests/UNITTESTS/double dir

Move events stubs headers to the mbed-os/events/tests/UNITTESTS/double
directory and update double CMake to include stub headers

rajkan01 committed:
CMake: unit-tests: Make events stub only depend on headers it uses

Previously the events stub library depended on `mbed-headers`, which
is a collection of all available headers in mbed-os. To make it easier
to separate the library, only depend on the headers we're using.

LDong-Arm reviewed (changes requested):
Similar to the other unit tests stubs PRs, maybe we can also move mbed-headers-events from UNITTESTS/stubs/CMakeLists.txt into events/tests/UNITTESTS/doubles/CMakeLists.txt?

And files in UNITTESTS/fakes/events/ can be moved/combined into events/tests/UNITTESTS/doubles/ too - fakes are a type of doubles too.

LDong-Arm commented on UNITTESTS/stubs/CMakeLists.txt:
```diff
@@ -101,7 +91,6 @@ target_include_directories(mbed-stubs-headers
-add_subdirectory(events)
 add_subdirectory(platform)
 add_subdirectory(storage)
 
```
(Referring to the lines below)
Shall we remove mbed-stubs-events from mbed-stubs, and let tests explicitly link mbed-stubs-events? Similar to what's been done to the headers - only link what is used.
Once all component stubs have been refactors, it'd be nice to have no global mbed-stubs but only component stubs eventually.

LDong-Arm replied:
Oh sorry this duplicates my comment from a few hours ago - I was re-checking the platforms PR (#14773) and just added the same comment to all other PRs 😅

rajkan01 replied:
I end up with a build issue, and we will do further cleanup or removing the complete mbed-os/UNITTESTS directory after we moved everything out

rajkan01 committed:
Remove trailing space

rajkan01 committed:
CMake: Move events fakes into events double dir

Previously events fakes as part of UNITTESTS/fakes, this PR moves
events fakes to events double directory to make events stubs
to be self-contained.

LDong-Arm reviewed (approved):
LGTM
